### PR TITLE
Prefill the payment form input with github handle

### PR DESCRIPTION
**Pull request by AnthonyBuisset**
closes E-282

### CODE REVIEW

File: frontend/src/pages/ProjectDetails/PaymentActions/PaymentForm/hooks/useIsGithubLoginValid/index.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { gql } from "@apollo/client";
```
Comment (Bernardstanislas) [inline frontend/src/pages/ProjectDetails/PaymentActions/PaymentForm/hooks/useIsGithubLoginValid/index.tsx:1]: Move it to the shared `hooks` folder

File: frontend/src/hooks/useHasuraQuery.tsx [hunk 2]
```diff
@@ -18,6 +20,14 @@ export const useHasuraQuery = <T, V = OperationVariables>(
   return useQuery(query, merge(options, { context: { headers: { "X-Hasura-Role": role } } }));
 };
 
+export const useHasuraLazyQuery = <T, V = OperationVariables>(
```
Comment (Bernardstanislas) [inline frontend/src/hooks/useHasuraQuery.tsx:23]: We faced a lot of troubles with @oscarwroche with the lazy queries, especially when loging out/in, is everything running smooth for you?

File: frontend/src/components/ContributorsTable/index.tsx [hunk 3]
```diff
@@ -158,11 +163,14 @@ const renderContributors = (contributors: Contributor[], isProjectLeader: boolea
...
           <div
             onClick={() => {
-              !isSendingNewPaymentDisabled &&
+              if (!isSendingNewPaymentDisabled) {
                 dispatch({
                   type: ProjectDetailsActionType.SelectPaymentAction,
                   selectedPaymentAction: PaymentAction.Send,
                 });
+                query.set("githubUserLogin", contributor.login);
+                navigate({ ...location, search: `?${query.toString()}` });
```
Comment (Bernardstanislas) [inline frontend/src/components/ContributorsTable/index.tsx:172]: I think we should leverage more React router for this, see https://reactrouter.com/en/main/route/route#optional-segments

**Review by Bernardstanislas — CHANGES_REQUESTED** [review]

File: frontend/src/hooks/useHasuraQuery.tsx (hunk 2 above)
Comment (AnthonyBuisset) [reply frontend/src/hooks/useHasuraQuery.tsx:23]: so far so good

File: frontend/src/components/ContributorsTable/index.tsx (hunk 3 above)
Comment (AnthonyBuisset) [reply frontend/src/components/ContributorsTable/index.tsx]: will check !

File: frontend/src/pages/ProjectDetails/PaymentActions/PaymentForm/hooks/useIsGithubLoginValid/index.tsx (hunk 1 above)
Comment (AnthonyBuisset) [reply frontend/src/pages/ProjectDetails/PaymentActions/PaymentForm/hooks/useIsGithubLoginValid/index.tsx:1]: done

File: frontend/src/components/ContributorsTable/index.tsx (hunk 3 above)
Comment (AnthonyBuisset) [reply frontend/src/components/ContributorsTable/index.tsx]: Looking at the documentation, using the location state is even better IMO. 
It is persisted during a window.reload but not part of the URL.
I think it fits really nicely here as the login is neither a search param, neither a nested route (like `project/:projectId`)

**Review by Bernardstanislas — APPROVED** [review]

**Pull request merged**